### DISCUSSION TITLE
INFO Required : python/java dataproc_templates GCSTOJDBC, how to pass SSL certificate details

### DISCUSSION THREAD

**Discussion by vinoth5490** [GoogleCloudPlatform/dataproc-templates#812] [Q&A] [answered]
Hi Team,

We are using GCSTOJDBC Template to load data into Postgres, need to know how to pass the Postgres SSL details (client-cert.pem,private-key.pem,server-cert.pem) in the template such that we can overcome the current issue which we are facing.
**Exception in thread "main" org.postgresql.util.PSQLException: FATAL: connection requires a valid client certificate**

Could you please let us know how we can pass those SSL details in GCSTOJDBC template.

**Comment by nj1973** [comment, marked as answer]
Hi Vinoth,

Assuming you are running the template from the command line then you should be able to pass the TLS files through to Dataproc Serverless using the FILES environment variable.

If you check `bin/start.sh` you should hopefully see this code which shows that values in `$FILES` will be passed into `gcloud` via the `--files` option.
```
if [ -n "${FILES}" ]; then
  OPT_FILES="--files=${FILES}"
fi
```

You would then need to pass an appropriate JDBC connection string via the `--gcs.jdbc.output.url` template option.

Please give this a try and let me know how you get on, or if I misunderstood the problem please provide further details.

Thanks
